layer/ro_layer.go

```diff
@@ -120,13 +120,14 @@ func storeLayer(tx MetadataTransaction, layer *roLayer) error {
 	if err := tx.SetCacheID(layer.cacheID); err != nil {
 		return err
 	}
-	if layer.parent != nil {
-		if err := tx.SetParent(layer.parent.chainID); err != nil {
+	// Do not store empty descriptors
+	if layer.descriptor.Digest != "" {
```

Comment:
I'm not sure that I like using concrete descriptors rather than pointers, and treating an empty digest field as an empty descriptor, but I'm also not sure the alternative would be better.

Main concern is that you can't specify just a media type, for example.